### PR TITLE
Remove restriction on extra keys when coercing Map/Object to struct

### PR DESCRIPTION
**Pull request by jdidion**
### Checklist
- [x] Pull request details were added to CHANGELOG.md
- [x] Valid examples WDL's were added or updated to the SPEC.md (see the [guide](https://github.com/openwdl/wdl-tests/blob/main/docs/MarkdownTests.md) on writing markdown tests)

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -1442,10 +1442,11 @@ Example output:
 
 🗑 It is also possible to assign an `Object` or `Map[String, X]` value to a `Struct` declaration. In the either case:
 
-* The `Object`/`Map` must not have any members that are not declared for the struct.
 * The value of each object/map member must be coercible to the declared type of the struct member.
 * The `Object`/`Map` must at least contain values for all of the struct's non-optional members.
 
+Keys in the map/object that do not match any struct members are ignored.
```
Comment (markjschreiber) [inline SPEC.md]: Is the cast destructive? For example if I cast a Map to a struct do the keys from the map that don't belong to the struct disappear from the Map or are they just inaccessible from the Struct pointer? What if I cast back again? Do they re-appear?

**Review by markjschreiber — CHANGES_REQUESTED** [review]
I think it would be good to have more clarity on what happens to the keys (and pairs) that don't match.

File: SPEC.md (hunk 1 above)
Comment (jdidion) [reply SPEC.md]: The new struct value is a copy, so it is not destructive in that sense. If you cast the Map to a struct without first saving the Map to a variable then it's gone. Casting from struct back to Map will not restore those extra members.

Comment (jdidion) [reply SPEC.md]: It's a good point though - the spec is not clear on whether type conversion is destructive. I'll add some clarification in that seciton.

**Review by markjschreiber — APPROVED** [review]

File: SPEC.md (hunk 1 above)
Comment (illusional) [reply SPEC.md]: Maybe saying _Only keys in the map / object that match struct members are stored in the struct_. But agree, the example Map -> Struct -> Map will not have the original fields.

**Review by illusional — APPROVED** [review]

**Pull request merged**